### ISSUE TITLE
Critical CSS blocks interactivity, shows broken image, causes jank in Luma

### ISSUE THREAD

**Issue by Leland** [magento/magento2#28913]
### Preconditions (*)
Magento >2.3.4
Magento 2.4-develop

### Steps to reproduce (*)
1. With Luma active on Magento >2.3.4, [enable Critical CSS](https://devdocs.magento.com/guides/v2.4/frontend-dev-guide/css-topics/css-critical-path.html):
    `bin/magento config:set dev/css/use_css_critical_path 1`

### Expected result (*)
* Inlined styles produces believable reproduction of site until full styles are loaded in
* Site remains somewhat interactive during loading process
* Luma critical CSS is best in class to give developers an excellent reference point

### Actual result (*)
* Mandatory loader mask completely blocks interaction with site until stylesheets are loaded in, negating much of the benefits to having critical CSS
* Loader mask spinner is an image, which is a lower resource priority than the preloaded CSS. Therefore, during the time in which critical CSS could be seen, the user sees a half-built site with a gray overlay preventing clicks, and a broken image reference placed at the top of the screen
* Finally, [Luma's critical CSS](https://github.com/magento/magento2/blob/ba25897c151b4f33505a8fe91aa1942bb6e71330/app/design/frontend/Magento/luma/web/css/critical.css), far from being best in class, is fundamentally too different from the final site to be of benefit. It actually _introduces_ jank, rather than reducing it.
    * The biggest example right now is that the inlined styles are using browser font defaults, causing the sans-serif theme to first render in Times New Roman. 
    * Since we're now exposed to the header before images have even loaded, the Luma logo should be moved to an SVG so that it's a part of the DOM and thus available for critical styles
    * Some – any – PageBuilder styles would be great here. As it stands, CMS pages render on critical styles as just blobs of alt text strung together.

All told, here is what the default critical CSS experience on Magento 2.3.5's Luma looks like:

![MicrosoftTeams-image (5)](https://user-images.githubusercontent.com/714017/85968300-bae2a200-b992-11ea-8440-8fec5ac4ce6c.png)

Compare to the final loaded result:

![image](https://user-images.githubusercontent.com/714017/85968322-ce8e0880-b992-11ea-80cf-949d45b5d6b7.png)

---
Please provide [Severity](https://devdocs.magento.com/guides/v2.3/contributor-guide/contributing.html#backlog) assessment for the Issue as Reporter. This information will help during Confirmation and Issue triage processes.

- [ ] Severity: **S0** _- Affects critical data or functionality and leaves users without workaround._
- [ ] Severity: **S1** _- Affects critical data or functionality and forces users to employ a workaround._
- [ ] Severity: **S2** _- Affects non-critical data or functionality and forces users to employ a workaround._
- [ ] Severity: **S3** _- Affects non-critical data or functionality and does not force users to employ a workaround._
- [x] Severity: **S4** _- Affects aesthetics, professional look and feel, “quality” or “usability”._

**Comment by engcom-Alfa** [comment]
The issue is still reproducible on fresh 2.4-develop

**_Preconditions:_**
1. Sample Data;
2. Enabled Critical Path (`bin/magento config:set dev/css/use_css_critical_path 1`);
3. Go to Storefront -> **Men** category (for example);

**Actual Result:** :heavy_multiplication_x: 

![Screenshot from 2021-01-20 14-58-10](https://user-images.githubusercontent.com/51679138/105178084-3e7efe80-5b30-11eb-8678-2aa6eec99810.png)

Compare to the final loaded result:

![Screenshot from 2021-01-20 14-58-25](https://user-images.githubusercontent.com/51679138/105178113-476fd000-5b30-11eb-8d2a-4593d31dcc4c.png)

**Comment by magento-engcom-team** [comment]
:white_check_mark: Confirmed by @engcom-Alfa 
 Thank you for verifying the issue. Based on the provided information internal tickets `MC-40520` were created

**Issue Available**: @engcom-Alfa, _You will be automatically unassigned. Contributors/Maintainers can claim this issue to continue. To reclaim and continue work, reassign the ticket to yourself._

**Comment by mrtuvn** [comment]
@magento give me 2.4-develop instance

**Comment by mrtuvn** [comment]
I think css critical not help much in this case when magento site current have so much requests js file. If you inspect source you will see a lot js file loading and it's may blocking page render. Seem when enable critical some other source have lower prioritize. Load later. Too much js delay time for eye notices jank visually

**Comment by Leland** [comment]
Yes, there are other things that could be moved off the critical path. That doesn't mean that critical CSS is pointless though, and, besides, this issue is about the fact that the critical CSS in Luma is _not good_. This repo should have an example of a really well-made critical CSS file.

**Comment by mrtuvn** [comment]
Seem we need extra work with critical for magento luma. From the docs guide i see some approach with penthouse but I'm not sure which is correct way

**Comment by mrtuvn** [comment]
https://github.com/magento/magento2/pull/31463#issuecomment-758793495

**Comment by Leland** [comment]
For this issue, we could even handwrite some manual critical CSS just as an example.

In my org, we are using Penthouse to generate the critical CSS, however. It would be an awesome thing to create a Grunt task that handles this, though that's totally unnecessary to just close this issue!

**Comment by roman-mer** [comment]
> For this issue, we could even handwrite some manual critical CSS just as an example.
> 
> In my org, we are using Penthouse to generate the critical CSS, however. It would be an awesome thing to create a Grunt task that handles this, though that's totally unnecessary to just close this issue!

Can you share your Penthouse setup in Magento 2?

**Comment by Leland** [comment]
@roman-mer sure - we're using Azure to handle deployments. We built a Penthouse node script that runs during the deployment that crawls through a headless Chrome instance and visits a specified number of URLs (homepage, PDP, etc). We then have some Regex that removes selectors that match to reduce the overall size. And finally, we have some manual CSS that is appended to it, which then produces a single critical.css file. It's very much in alpha state (and proprietary) so no code to share, sadly.

**Comment by roman-mer** [comment]
Someone might share a Penthouse setup in Grunt (Magento 2)
I found such a customization recommendation, but so far I have not been able to implement it:

add new task
grunt.initConfig({
  penthouse: {
    extract : {
        outfile : '.tmp/out.css',
        css : './dist/css/full.css',
        url : 'http://localhost:9000',
        width : 1300,
        height : 900,
        skipErrors : false // this is the default
    },
  },
});

**Comment by mrtuvn** [comment]
i think there are not common results for all here. You have to build your own tools to meet your requirements
Build process deploy like Leland describe above. Css will change not only by the page you visit but also the  viewport related (width height). Height 200 will be difference with Height 900
Default luma critical would not much help for you since styles quite big with we  have alot styles and merged to 2 files (styles-l, styles-m). Almost of them (90%) is not all you need for specific page

**Comment by mrtuvn** [comment]
@Leland Can you share how big size of your critical.css in your org? My site generate file around ~ 300KB
Quite big for single critical is it?

**Comment by Leland** [comment]
@mrtuvn depends on the site. For this issue, I'd recommend staying in the 15-50kb range, however. Increasing every page's document size by 300kb will cause  Luma's First Contentful Paint time to increase significantly.

**Comment by Leland** [comment]
This is a confirmed issue, and should not ever be marked stale IMO… perhaps stale bot’s settings should be tweaked?

**Comment by mrtuvn** [comment]
Hi @Leland https://github.com/magento/magento2/pull/33084/files we already have PR for such case
@ihor-sviziev should we add exempt labels also for confirmed

**Comment by ihor-sviziev** [comment]
@mrtuvn let's discuss it with other maintainers

**Comment by engcom-Bravo** [comment]
Hi @Leland,

Thanks for your reporting and collaboration.

We have verified the issue in Latest Mganeto instance and the issue is no more reproducible.Kindly refer the screenshots.

<img width="1391" alt="Screenshot 2025-01-13 at 10 09 51 am" src="https://github.com/user-attachments/assets/00609e9b-6b5b-4235-8580-957dbd77ca53" />
<img width="1391" alt="Screenshot 2025-01-13 at 10 10 04 am" src="https://github.com/user-attachments/assets/3447a477-6e35-4e2b-b168-fff129418293" />


Hence we are closing this issue.

Thanks.
